datahub-web-react/src/App.tsx

```diff
@@ -25,18 +29,24 @@ const client = new ApolloClient({
 });
 
 const App: React.VFC = () => {
+    // Register our entities!
+    const entityRegistry = new EntityRegistry();
+    entityRegistry.register(new DatasetEntity());
```

Comment:
This is not a blocker. I just think this part of "register" code can be:
1. auto-generate at compile time
2. execute inside EntityRegistry class, at construction time.

Reply:
Either way this should be done in a useEffect() so that we dont register our entities multiple times

Reply:
1. Agree we can do some magic to ensure we don't forget to register. 
2. We cannot do this, because it will create a cyclic dependency. (EntityRegistry --> DatasetEntity --> EntityRegistry) 

Agree we should do this in useEffect to avoid this occurring on every render! Good call out!

Reply:
+1 what woody said

Reply:
I would try to make these dynamic/config driven

Reply:
I don't see a strong argument for these being configurable. The registry is not implying that you can just enable / disable particular entities with no consequences. This is because often multiple entity types are implicitly related. 

For example, if you disabled "User" much of the Dataset's functionality in the UI would also stop working (think "ownership" tab. These dependencies will typically manifest in the entity details page. 

Instead, we will have per-entity configuration, where you can enable / disable more granular features like search, browse, particular tabs in the details page, etc.

Reply:
Update: i've tried #2 above but we have a dependency cycle, where EntityRegistry --> DatasetEntity --> EntityRegistry, as the DatasetEntity will sometimes defer to the registry when doing things like rendering the "owners" tab of the profile. Going to just memoize this logic for now. 

Will add a TODO about making it more dynamically configurable!

Reply:
Resolving for now